sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -638,6 +645,17 @@ class Analyzer(
           }
           val newAggregateExpressions = a.aggregateExpressions ++ missingAttrs
           a.copy(aggregateExpressions = newAggregateExpressions)
+        case g: Generate =>
+          if (g.outer && !g.join) {
```

Comment:
Yeah, true. However, if that happens, we will see the errors. Thus, I put it for ensuring it will never happen. Or should I just put `assert`?